src/crimson/net/Socket.cc

```diff
@@ -67,4 +67,15 @@ seastar::future<bufferlist> Socket::read(size_t bytes)
     });
 }
 
+seastar::future<seastar::temporary_buffer<char>>
+Socket::read_exactly(size_t bytes) {
+  return in.read_exactly(bytes)
+    .then([this, bytes](auto buf) {
+      if (buf.size() < bytes) {
```

Comment:
could be more specific, like
```c++
if (buf.empty()) {
  throw std::system_error(make_error_code(error::read_eof));
}
```

Reply:
Please refer to 
https://github.com/scylladb/scylla/blob/master/sstables/sstables.cc#L178-L189

Maybe I should add some comments here.

Reply:
I see, the implementation of `read_exactly()` will return an empty buffer for short reads. Thanks.

Reply:
fixed.